File: swift/Programmers/Programmers/lev2_덧칠하기.swift
//
//  lev2_덧칠하기.swift
//  Programmers
//
//  Created by ryu hyunsun on 2023/03/04.
//

import Foundation

func solution(_ n:Int, _ m:Int, _ section:[Int]) -> Int {
    var answer = 0
    var painted = 0
    for i in 0..<section.count {
        if section[i] > painted {
            painted = section[i] + m - 1
            answer += 1
        }
    }
    
    
    return answer
}

File: python/알고리즘/pro/lev2_덧칠하기.py
def solution(n, m, section):
    answer = 0
    painted = 0
    for i in range(len(section)):
        # 마지막으로 페인트 칠 된 곳을 확인해서, 그것보다 작으면 칠 안된것이니까 갱신.
        if painted < section[i]:
            painted = section[i] + m - 1
            answer += 1
    return answer